### PR TITLE
integrated LLMs for creating distractors

### PR DESCRIPTION
**Pull request by n4-t3**
integrated LLMs for creating distractors in add options page:)

### CODE REVIEW

File: src/pages/api/getGPTDistractor.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  async ({ qStem, qLearningObjective }) => {
+    const promptQuestion = `Create 3 distractors separated by a comma for the following question ${qStem}; its learning objective is ${qLearningObjective}, reply only with the distractors.`
+
+    const openAIKey = Env.OPEN_AI_KEY
+    const openAI = new OpenAIApi(
+      new Configuration({
+        apiKey: openAIKey,
+      })
+    )
+    const openAIResponse = await openAI.createChatCompletion({
```
Comment (jhw123) [inline src/pages/api/getGPTDistractor.ts:24]: I suggest to create a new service (under `/server/services`) for methods related to generative AI

Comment (n4-t3) [reply src/pages/api/getGPTDistractor.ts]: > I suggest to create a new service (under `/server/services`) for methods related to generative AI

created the service here https://github.com/kixlab/kuiz/pull/81/commits/03082054d8bbd6da2b4927ab4910e49ad74f22d3 :)

File: src/server/services/openAI.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+interface OpenAIData {
+  apiKey: string
+  model: 'gpt-3.5-turbo' | 'gpt-4'
+  role: 'user' | 'assistant' | 'system'
+  content: string
+}
+
+class OpenAIService {
+  async create({ apiKey, model, role, content }: OpenAIData) {
```
Comment (jhw123) [inline src/server/services/openAI.ts:11]: `apiKey` does not need to be passed every time. Set it as a private attribute of the class and initialize it with `Env.OPEN_AI_KEY`

File: src/server/services/openAI.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    const openAI = new OpenAIApi(
+      new Configuration({
+        apiKey,
+      })
+    )
```
Comment (jhw123) [inline src/server/services/openAI.ts:16]: make this run only once at class initialization

**Review by jhw123 — CHANGES_REQUESTED** [review]

File: src/server/services/openAI.ts (hunk 2 above)
Comment (n4-t3) [reply src/server/services/openAI.ts]: > `apiKey` does not need to be passed every time. Set it as a private attribute of the class and initialize it with `Env.OPEN_AI_KEY`

That is right! I have updated it here https://github.com/kixlab/kuiz/pull/81/commits/5bb867055c576b9f58ca13339587e8a6a4f79c1f :)

File: src/server/services/openAI.ts (hunk 3 above)
Comment (n4-t3) [reply src/server/services/openAI.ts]: > make this run only once at class initialization

I have updated it here https://github.com/kixlab/kuiz/pull/81/commits/b9185fd1247f2d9416bd6d83fe9b0497c58d26ed :)

**Review by jhw123 — APPROVED** [review]

**Pull request merged**